overlord/configstate/configcore/proxy_test.go

```diff
@@ -159,10 +155,11 @@ func (s *proxySuite) TestConfigureProxyStore(c *C) {
 		"timestamp":   time.Now().Format(time.RFC3339),
 	}, nil, "")
 	c.Assert(err, IsNil)
-	s.state.Lock()
-	err = assertstate.Add(s.state, stoAs)
-	s.state.Unlock()
-	c.Assert(err, IsNil)
+	func() {
```

Comment:
Same question as above about a similar pattern, a (short) comment would be nice why this pattern is used.